File: mobile/lib/core/models/rating.dart
class Rating {
  final String id;
  final String clientId;
  final String delivererId;
  final String orderId;
  final int rating;

  Rating({
    required this.id,
    required this.clientId,
    required this.delivererId,
    required this.orderId,
    required this.rating,
  });

  factory Rating.fromJson(Map<String, dynamic> json) {
    return Rating(
      id: json['id'],
      clientId: json['client']?['id'] ?? json['clientId'],
      delivererId: json['deliverer']?['id'] ?? json['delivererId'],
      orderId: json['orderId'],
      rating: json['rating'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'clientId': clientId,
      'delivererId': delivererId,
      'orderId': orderId,
      'rating': rating,
    };
  }
}

File: models/Rating.js
const { Model, DataTypes } = require('sequelize')
const ratingDto = require('../mongo-models/dtos/rating-dto')
const operations = require('../mongo-models/dtos/operations')

module.exports = function (connection) {
    class Rating extends Model {
        static associate(db) {
            Rating.belongsTo(db.User, {
                as: 'client',
                foreignKey: 'clientId',
            })
            db.User.hasMany(Rating, { as: 'ratings', foreignKey: 'clientId' })

            Rating.belongsTo(db.Deliverer, {
                as: 'deliverer',
                foreignKey: {
                    name: 'delivererId',
                    allowNull: true,
                },
            })
            db.Deliverer.hasMany(Rating, {
                as: 'ratings',
                foreignKey: {
                    name: 'delivererId',
                    allowNull: true,
                },
            })

            Rating.belongsTo(db.Order, {
                as: 'order',
                foreignKey: { name: 'orderId', allowNull: true },
            })
        }

        static addHooks(db) {
            Rating.addHook('afterCreate', (rating) => {
                ratingDto(rating.id, db.Rating)
            })

            Rating.addHook('afterUpdate', (rating) => {
                ratingDto(rating.id, db.Rating, operations.update)
            })

            Rating.addHook('afterDestroy', (rating) => {
                ratingDto(rating.id, db.Rating, operations.delete)
            })
        }
    }

    Rating.init(
        {
            id: { type: DataTypes.UUID, primaryKey: true },
            rating: {
                type: DataTypes.INTEGER,
                allowNull: false,
            },
        },
        {
            sequelize: connection,
            tableName: 'rating',
        }
    )

    return Rating
}
